### PR TITLE
Added UserUpdate endpoint to Quarterdeck

### PR DESCRIPTION
**Pull request by pdamodaran**
### Scope of changes

Added UserUpdate endpoint to Quarterdeck.  This allows for users with collaborators:edit permissions to edit a user's information.  This currently does not have functionality to update a user's email.  That work will be completed in a follow on story.

Fixes SC-13266

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [ ]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [ ]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/quarterdeck/api/v1/api.go [hunk 1]
```diff
@@ -32,6 +31,9 @@ type QuarterdeckClient interface {
 
 	// Project Resource
 	ProjectCreate(context.Context, *Project) (*Project, error)
+
+	// Users Resource
+	UserUpdate(context.Context, *User) error
```
Comment (pdeziel) [inline pkg/quarterdeck/api/v1/api.go:36]: Do you think that `UserUpdate` should return a `*User` to be in line with the other API endpoints?

**Review by pdeziel — COMMENTED** [review]
This looks really good, I just had a few questions and suggestions but they shouldn't be too bad to implement.

Can I also request to add this endpoint to the mock at `pkg/quarterdeck/mock/quarterdeck.go`, following the pattern of the other endpoints?

File: pkg/quarterdeck/db/models/users_test.go [hunk 2]
```diff
@@ -524,3 +524,41 @@ func (m *modelTestSuite) TestUserPermissions() {
...
+
+	// failure to pass in valid orgID returns error
+	user.Name = "Sarah Fisher"
+	err = user.UserUpdate(ctx, orgID)
+	require.Equal("object not found in the database", err.Error())
+
+	// passing an orgID that's different from the user's organization results in an error
+	orgID = ulid.MustParse("01GQZAC80RAZ1XQJKRZJ2R4KNJ")
+	err = user.UserUpdate(ctx, orgID)
+	require.Equal("object not found in the database", err.Error())
```
[the remark this answers is not in the document]
Comment (pdamodaran) [reply pkg/quarterdeck/db/models/users_test.go:562]: updated - thank you!

File: pkg/quarterdeck/api/v1/api.go [hunk 3]
```diff
@@ -275,3 +277,25 @@ type OpenIDConfiguration struct {
...
+// Users Resource
+// ===========================================================================
+
+// TODO: add Email
+type User struct {
+	UserID ulid.ULID `json:"user_id"`
+	Name   string    `json:"name"`
+}
+
+// TODO: validate Email
```
[the remark this answers is not in the document]
Comment (pdamodaran) [reply pkg/quarterdeck/api/v1/api.go:296]: yep - it's on my to-do list.

**Review by pdeziel — APPROVED** [review]
LGTM!

**Pull request merged**